openapi-cli/src/main/java/io/ballerina/openapi/generators/schema/TypeGeneratorUtils.java

```diff
@@ -155,17 +161,36 @@ public static TypeDescriptorNode getNullableType(Schema schema, TypeDescriptorNo
             MetadataNode metadataNode = createMetadataNode(documentationNode, createEmptyNodeList());
             if (required != null) {
                 if (!required.contains(field.getKey().trim())) {
-                    recordFieldNode = NodeFactory.createRecordFieldNode(metadataNode, null,
-                            fieldTypeName, fieldName, createToken(QUESTION_MARK_TOKEN), createToken(SEMICOLON_TOKEN));
+                    if (field.getValue().getDefault() != null) {
+                        Token defaultValue;
+                        if ((field.getValue().getDefault()).getClass() == String.class) {
+                            defaultValue = AbstractNodeFactory.createIdentifierToken("\"" + 
+                            field.getValue().getDefault().toString() + "\"");
+                        } else {
+                            defaultValue = AbstractNodeFactory.createIdentifierToken
+                            (field.getValue().getDefault().toString());
+                        }
+                        ExpressionNode expressionNode = createRequiredExpressionNode(defaultValue);
+                        recordFieldWithDefaultValueNode = NodeFactory.createRecordFieldWithDefaultValueNode
```

Comment:
Shall we initialize `recordFieldWithDefaultValueNode` node here since there is no other usage of it other than here